### PR TITLE
Map value of `azure_client_id` to `managed-identity-client-id`

### PR DESCRIPTION
**Pull request by backwind1233**
by default the value of azure_client_id will be assigned to  "credential.client-id",
we also need to assign the value to `credential.managed-identity-client-id"`.

### CODE REVIEW

**Comment by backwind1233** [comment]
@saragluna please help review this PR

**Comment by stliu** [comment]
this only take a env value but not properties

considering the following configuration

`credential.client-id=xxxx`, will this value be assigned to the `credential.managed-identity-client-id` as well?

I think the it's better to impl something like :

`credential.managed-identity-client-id = ${credential.client-id}` at autoconfiguration's property binding stage

**Comment by saragluna** [comment]
@backwind1233  will you address Strong's comment?

File: sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/context/AzureGlobalConfigurationEnvironmentPostProcessorTest.java [hunk 1]
```diff
@@ -140,6 +140,34 @@ void azureCoreEnvShouldBindCorrect() {
...
+        assertEquals("client-id-from-env", globalProperties.getCredential().getClientId());
+        assertEquals("client-id-from-env", globalProperties.getCredential().getManagedIdentityClientId());
+        assertEquals(null, globalProperties.getCredential().getUsername());
+    }
+
+    @Test
+    void azureManagedIdentityClientIdFromUserConfig() {
+        Properties properties = new Properties();
+        properties.put(PROPERTY_AZURE_CLIENT_ID, "client-id-from-env");
+        properties.put(AzureGlobalProperties.PREFIX + ".credential.managed-identity-client-id",
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/context/AzureGlobalConfigurationEnvironmentPostProcessorTest.java:160]: the value better follow the same pattern, you can use the custom-managed-identity-client-id

**Review by saragluna — CHANGES_REQUESTED** [review]

**Review by saragluna — APPROVED** [review]
LGTM.

**Pull request merged**